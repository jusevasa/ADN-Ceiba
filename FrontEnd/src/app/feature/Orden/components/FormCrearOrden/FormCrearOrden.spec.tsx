import * as React from 'react';

import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';

import { FormCrearOrden } from './';
import { setTextEvent } from 'app/shared/utils/test';

describe('FormCrearProducto test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof FormCrearOrden> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      formTitle: 'Crear orden',
      onSubmit: stub(),
    };
    componentWrapper = render(<FormCrearOrden {...componentProps} />);
  });

  it('should render component', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should fail on submit all fields missing', async () => {
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(2);
    expect(spans[0].textContent).toBe('El campo fecha de entrega es requerido.');
    expect(spans[1].textContent).toBe('El campo hora de entrega es requerido.');
  });


});
