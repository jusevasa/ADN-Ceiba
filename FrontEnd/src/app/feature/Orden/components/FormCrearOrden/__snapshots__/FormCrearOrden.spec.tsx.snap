// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCrearProducto test should render component 1`] = `
<div>
  <form
    class=""
  >
    <h2>
      Crear orden
    </h2>
    <div
      class="row"
    >
      <div
        class="col-xl-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="idCoordinador"
          >
            Id Coordinador
          </label>
          <input
            class="form-control"
            id="idCoordinador"
            max="4"
            min="1"
            name="idCoordinador"
            placeholder="Id Coordinador"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="col-xl-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="idRepartidor"
          >
            Id Repartidor
          </label>
          <input
            class="form-control"
            id="idRepartidor"
            max="8"
            min="1"
            name="idRepartidor"
            placeholder="Id Repartidor"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="col-xl-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="idFechaEntrega"
          >
            Fecha Entrega
          </label>
          <input
            class="form-control"
            id="idFechaEntrega"
            name="fechaEntrega"
            placeholder="Fecha Entrega"
            type="date"
            value=""
          />
        </div>
      </div>
      <div
        class="col-xl-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="idHoraEntrega"
          >
            Hora Entrega
          </label>
          <input
            class="form-control"
            id="idHoraEntrega"
            name="horaEntrega"
            placeholder="Fecha Entrega"
            type="time"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      class="w-100 btn btn-secondary"
      type="submit"
    >
      Crear
    </button>
  </form>
</div>
`;
